lib/util/http/retry-after.ts

```diff
@@ -0,0 +1,102 @@
+import { setTimeout } from 'timers/promises';
+import { RequestError } from 'got';
+import { DateTime } from 'luxon';
+import { logger } from '../../logger';
+import { parseUrl } from '../url';
+import type { Task } from './types';
+
+const hostBlocks = new Map<string, Promise<void>>();
+
+const maxRetries = 2;
+
+export async function wrapWithRetry<T>(
+  task: Task<T>,
+  url: string,
+  getRetryAfter: (err: unknown) => number | null,
+  maxRetryAfter: number,
+): Promise<T> {
+  const key = parseUrl(url)?.host ?? /* istanbul ignore next */ url;
+
+  let retries = 0;
+  for (;;) {
+    try {
+      const delayPromise = hostBlocks.get(key);
+      if (delayPromise) {
+        await delayPromise;
+        hostBlocks.delete(key);
```

Comment:
see below, Ed should check if it's our delay and await the new if it's not ours. see below

Reply:
Actually, now I think it's better not to share delay object at all, otherwise it's too complex

Reply:
Or, accumulate delays using `Promise.all`